test/mongodb.test.js

```diff
@@ -3073,6 +3073,60 @@ describe('mongodb connector', function() {
     module.generateMongoDBURL.should.be.an.instanceOf(Function);
   });
 
+  describe('Test generateMongoDBURL function', function() {
+    var module = require('../');
+    context('should return correct mongodb url ', function() {
+      it('when only passing in database', function () {
+        var options = {
+          database: 'fakeDatabase'
+        };
+        module.generateMongoDBURL(options).should.be.eql('mongodb://127.0.0.1:27017/fakeDatabase');
+      });
+      it('when protocol is mongodb and no username/password', function () {
+        var options = {
+          protocol: 'mongodb',
+          hostname: 'fakeHostname',
+          port: 9999,
+          database: 'fakeDatabase'
+        };
+        module.generateMongoDBURL(options).should.be.eql('mongodb://fakeHostname:9999/fakeDatabase');
+      });
+      it('when protocol is mongodb and no username/password', function () {
+        var options = {
+          protocol: 'mongodb',
+          hostname: 'fakeHostname',
+          port: 9999,
+          database: 'fakeDatabase',
+          username: 'fakeUsername',
+          password: 'fakePassword'
+        };
+        module.generateMongoDBURL(options).should.be.eql('mongodb://fakeUsername:fakePassword@fakeHostname:9999/fakeDatabase');
+      });
+      it('when protocol is mongodb+srv and no username/password', function () {
+        var options = {
+          protocol: 'mongodb+srv',
+          hostname: 'fakeHostname',
+          port: 9999,
+          database: 'fakeDatabase'
+        };
+        // mongodb+srv url should not have the port in it
+        module.generateMongoDBURL(options).should.be.eql('mongodb+srv://fakeHostname/fakeDatabase');
+      });
+      it('when protocol is mongodb+srv and no username/password', function () {
```

Comment:
Name of test should be 'when protocol is mongodb+srv and username/password' since the
user name and password IS being passed in the options.

Reply:
Corrected to state it has username and password